plugins/integration/IntegrationProviderPlugin.php

```diff
@@ -34,6 +34,14 @@ public static function getEnums($baseEnumName = null)
 		return array();
 	}
 
+	/**
+	 * @return IIntegrationProvider
+	 */
+	public static function getProvider($providerType)

```

Comment:
Make this method abstract and implement it in each integration provider instead of calling the plugins manager.